src/Microsoft.ML.Data/Data/DataViewTypeManager.cs

```diff
@@ -113,6 +117,16 @@ internal static bool Knows(DataViewType dataViewType)
         /// <param name="typeAttributes">The <see cref="Attribute"/>s attached to <paramref name="type"/>.</param>
         public static void Register(DataViewType dataViewType, Type type, IEnumerable<Attribute> typeAttributes = null)
```

Comment:
>IEnumerable [](start = 74, length = 11)

It's not a part of this change, but I'm just wondering why is this an enumerable, and not just one attribute? What would be the scenario for having multiple attributes?
In the `SchemaDefinition.Create` method (SchemaDefinition.cs line 388) it seems like it is not possible to have more than one `DataViewTypeAttribute`, or am I missing something?

Also, how is this method intended to be used? From the sample, it seems that it is only used for implementing the abstract method `Register()` of `DataViewTypeAttribute`. If this is the case, shouldn't `Register()` be implemented in the base class by calling this method and have abstract methods/properties to get the arguments needed for it? This way the `Register` method here can be internal, which would be less confusing (at least to me :)).

/cc @eerhardt .

Reply:
@wschin - for the question. He originally added the API.

Since this is a public API, we probably can't change it since it has already shipped.

Reply:
I believe I have found the appropriate approach to how to address the API, please see the latest commit.